core/src/main/java/juzu/impl/common/Tools.java

```diff
@@ -194,7 +197,11 @@ public boolean hasNext() {
 
           @Override
           public T next() {
-            if (!hasNext()) {
+            if(next != null) {
```

Comment:
We have the 2 same piece of code in this if and in the else :
`T tmp = next;
next = null;
return tmp;`
Is it necessary to repeat it ?